lib/Parse/ParseDecl.cpp

```diff
@@ -803,14 +843,16 @@ Parser::parseDifferentiableAttribute(SourceLoc atLoc, SourceLoc loc) {
     diagnose(Tok, diag::attr_differentiable_expected_mode);
     return errorAndSkipToEnd();
   }
-  auto modeLoc = consumeToken(tok::identifier); // 'forward' or 'reverse'.
+  modeLoc = consumeToken(tok::identifier);
   // Parse comma.
   parseToken(tok::comma, diag::attr_expected_comma, "differentiable",
```

Comment:
Not sure whether you've seen my comments on the previous PR. Could we use `AttrName` here instead of spelling out "differentiable" ? similarly for the other instances of "differentiable".

Reply:
Thanks! Done in https://github.com/apple/swift/pull/17428/commits/913f70d6879a70e55305f90f5d5d2fdf4f593d5e.